packages/contract-wrappers/src/utils/transaction_encoder.ts

```diff
@@ -0,0 +1,293 @@
+import { schemas } from '@0xproject/json-schemas';
+import { EIP712Schema, EIP712Types, EIP712Utils } from '@0xproject/order-utils';
+import { Order, SignedOrder } from '@0xproject/types';
+import { BigNumber } from '@0xproject/utils';
+import _ = require('lodash');
+
+import { ExchangeContract } from '../contract_wrappers/generated/exchange';
+
+import { assert } from './assert';
+
+const EIP712_ZEROEX_TRANSACTION_SCHEMA: EIP712Schema = {
+    name: 'ZeroExTransaction',
+    parameters: [
+        { name: 'salt', type: EIP712Types.Uint256 },
+        { name: 'signerAddress', type: EIP712Types.Address },
+        { name: 'data', type: EIP712Types.Bytes },
+    ],
+};
+
+/**
+ * Transaction Encoder. Transaction messages exist for the purpose of calling methods on the Exchange contract
+ * in the context of another address. For example, UserA can encode and sign a fillOrder transaction and UserB
+ * can submit this to the blockchain. The Exchange context executes as if UserA had directly submitted this transaction.
+ */
+export class TransactionEncoder {
+    private _exchangeInstance: ExchangeContract;
+    constructor(exchangeInstance: ExchangeContract) {
+        this._exchangeInstance = exchangeInstance;
+    }
+    /**
+     * Encodes the transaction data for use with the Exchange contract.
+     * @param data The ABI Encoded 0x Exchange method. I.e fillOrder
+     * @param salt A random value to provide uniqueness and prevent replay attacks.
+     * @param signerAddress The address which will sign this transaction.
+     * @return An unsigned hex encoded transaction for use in 0x Exchange executeTransaction.
+     */
+    public getTransactionHex(data: string, salt: BigNumber, signerAddress: string): string {
+        const exchangeAddress = this._getExchangeContract().address;
+        const executeTransactionData = {
+            salt,
+            signerAddress,
+            data,
+        };
+        const executeTransactionHashBuff = EIP712Utils.structHash(
+            EIP712_ZEROEX_TRANSACTION_SCHEMA,
+            executeTransactionData,
+        );
+        const eip721MessageBuffer = EIP712Utils.createEIP712Message(executeTransactionHashBuff, exchangeAddress);
+        const messageHex = `0x${eip721MessageBuffer.toString('hex')}`;
+        return messageHex;
+    }
+    /**
+     * Encodes a fillOrder transaction.
+     * @param  signedOrder           An object that conforms to the SignedOrder interface.
+     * @param  takerAssetFillAmount  The amount of the order (in taker asset baseUnits) that you wish to fill.
+     * @return Hex encoded abi of the function call.
+     */
+    public fillOrder(signedOrder: SignedOrder, takerAssetFillAmount: BigNumber): string {
```

Comment:
Looks like these haven't been renamed to `getFillOrderTx`, etc... Please rename.